tests/test_esm_runscripts/test_namelist.py

```diff
@@ -22,16 +27,30 @@
 )
 
 
+class Capturing(list):
+    """Taken from https://stackoverflow.com/questions/16571150/how-to-capture-stdout-output-from-a-python-function-call"""
+    def __enter__(self):
+        self._stdout = sys.stdout
+        sys.stdout = self._stringio = StringIO()
+        return self
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio    # free up some memory
+        sys.stdout = self._stdout
+
+
 class TestNamelist(unittest.TestCase):
     """Test the various static methods of Namelist"""
 
     def setUp(self):
         """Prepares everything to might be needed"""
         self.config = yaml.safe_load(CONFIG_YAML)
         self.mconfig = self.config["echam"]
+        self._nml_name = self.mconfig["namelists"][0]
         self._thisrun_config_dir = tempfile.mkdtemp()
         self.mconfig["thisrun_config_dir"] = self._thisrun_config_dir
-        print(self.mconfig)
+        self._thisrun_namelist_path = os.path.join(self._thisrun_config_dir, self._nml_name)
+        shutil.copy2(TEST_NML_PATH, self._thisrun_namelist_path)
```

Comment:
Looking at this: we only test one echam namelist. Maybe it is better to test all namelists we provide? Then again, we are checking behaviour here, which should work the same no matter which one we give in. Perhaps it would even be better to make a "fake" namelist to check against.

Reply:
done